aws/resource_aws_api_gateway_request_validator.go

```diff
@@ -16,6 +17,20 @@ func resourceAwsApiGatewayRequestValidator() *schema.Resource {
 		Read:   resourceAwsApiGatewayRequestValidatorRead,
 		Update: resourceAwsApiGatewayRequestValidatorUpdate,
 		Delete: resourceAwsApiGatewayRequestValidatorDelete,
+		Importer: &schema.ResourceImporter{
+			State: func(d *schema.ResourceData, meta interface{}) ([]*schema.ResourceData, error) {
+				idParts := strings.Split(d.Id(), "/")
+				if len(idParts) != 2 || idParts[0] == "" || idParts[1] == "" {
+					return nil, fmt.Errorf("Unexpected format of ID (%q), expected REST-API-ID/REQUEST-VALIDATOR-ID", d.Id())
+				}
+				restApiID := idParts[0]
+				requestValidatorID := idParts[1]
+				d.Set("request_validator_id", requestValidatorID)
+				d.Set("rest_api_id", restApiID)
+				d.SetId(requestValidatorID)
```

Comment:
why not just call d.SetID and then the Read function? 🤔

Reply:
In this case, the read function needs the REST API ID as well as the request validator ID to perform the read, but the existing resource ID is missing the REST API ID. At some point we should adjust the resource ID to include the REST API ID so we can perform reads correctly from ID, but for now we can support import without adjusting other code.